src/modules/rover_pos_control/RoverPositionControl.cpp

```diff
@@ -286,11 +286,23 @@ RoverPositionControl::control_velocity(const matrix::Vector3f &current_velocity,
 
 		const float control_throttle = pid_calculate(&_speed_ctrl, desired_speed, x_vel, x_acc, dt);
 
+		//Constrain maximum throttle to mission throttle
 		_act_controls.control[actuator_controls_s::INDEX_THROTTLE] = math::constrain(control_throttle, 0.0f, mission_throttle);
 
-		const Vector3f desired_velocity = R_to_body * Vector3f(desired_local_velocity(0), desired_local_velocity(1),
-						  desired_local_velocity(2));
-		const float desired_theta = atan2f(desired_velocity(1), desired_velocity(0));
+		Vector3f desired_body_velocity;
+
+		if (pos_sp_triplet.current.velocity_frame == position_setpoint_s::VELOCITY_FRAME_BODY_NED) {
+			desired_body_velocity = Vector3f(desired_velocity(0), desired_velocity(1),
```

Comment:
Aren't these both Vector3fs?
```suggestion
			desired_body_velocity = desired_velocity;
```

Reply:
Addressed in the last commit!